File: src/main/java/yuuto/enhancedinventories/gui/GuiHandler.java
/*******************************************************************************
 * Copyright (c) 2014 Yuuto.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the GNU Public License v3.0
 * which accompanies this distribution, and is available at
 * http://www.gnu.org/licenses/gpl.html
 *
 * Contributors:
 * 	   cpw - src reference from Iron Chests
 * 	   doku/Dokucraft staff - base chest texture
 *     Yuuto - initial API and implementation
 ******************************************************************************/
package yuuto.enhancedinventories.gui;

import yuuto.enhancedinventories.client.GuiContainerDynamic;
import yuuto.enhancedinventories.tile.TileConnectiveInventory;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.world.World;
import cpw.mods.fml.common.network.IGuiHandler;

public class GuiHandler implements IGuiHandler{

	@Override
	public Object getServerGuiElement(int ID, EntityPlayer player, World world,
			int x, int y, int z) {
		if(ID == 0){
			TileEntity tile = world.getTileEntity(x, y, z);
			if(!(tile instanceof TileConnectiveInventory))
				return null;
			TileConnectiveInventory inv = (TileConnectiveInventory)tile;
			/*if(inv.getSizeInventory() <= 54){
				return new ContainerConnected(inv, player);
			}
			return new ContainerConnectedLarge(inv, player);*/
			return new ContainerDynamic(inv, player);
		}
		return null;
	}

	@Override
	public Object getClientGuiElement(int ID, EntityPlayer player, World world,
			int x, int y, int z) {
		if(ID == 0){
			TileEntity tile = world.getTileEntity(x, y, z);
			if(!(tile instanceof TileConnectiveInventory))
				return null;
			TileConnectiveInventory inv = (TileConnectiveInventory)tile;
			/*if(inv.getSizeInventory() <= 54){
				return new GuiContainerConnectedDyn(inv, player);
				//return new GuiContainerConnected(inv, player);
			}
			return new GuiContainerConnectedLarge(inv, player);*/
			return new GuiContainerDynamic(inv, player);
		}
		return null;
	}

}

File: src/main/scala/yuuto/enhancedinventories/gui/GuiHandler.scala
/**
 * @author Yuuto
 */
package yuuto.enhancedinventories.gui

import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.world.World;
import yuuto.enhancedinventories.client.gui.GuiContainerAutoAssembler;
import yuuto.enhancedinventories.client.gui.GuiContainerDynamic;
import yuuto.enhancedinventories.client.gui.GuiContainerPainter;
import yuuto.enhancedinventories.client.gui.GuiContainerWorkbench;
import yuuto.enhancedinventories.client.gui.GuiContainerWorktable;
import yuuto.enhancedinventories.tile.TileAutoAssembler;
import yuuto.enhancedinventories.tile.TilePainter;
import yuuto.enhancedinventories.tile.TileWorktable;
import yuuto.enhancedinventories.tile.traits.TInventorySimple;
import yuuto.enhancedinventories.tile.traits.TSecurable;
import cpw.mods.fml.common.network.IGuiHandler;

class GuiHandler extends IGuiHandler{

  override def getServerGuiElement(ID:Int, player:EntityPlayer, world:World,x:Int, y:Int, z:Int):Object={
    val tile:TileEntity = world.getTileEntity(x, y, z);
    ID match{
    //Open chest inventory
    case 0=>{
      if(!tile.isInstanceOf[TInventorySimple])
        return null;
      val inv:TInventorySimple = tile.asInstanceOf[TInventorySimple];
      return new ContainerDynamic(inv.getInventory(), player);
    }
    //Open a workbench inventory
    case 1 =>{
      return new ContainerWorkbench(player, world, x,y,z);
    }
    //Open Worktable inventory
    case 2 =>{
      if(!tile.isInstanceOf[TileWorktable])
        return null;
      return new ContainerWorktable(tile.asInstanceOf[TileWorktable], player);
    }
    //Open Auto Assembler inventory
    case 3 =>{
      if(!tile.isInstanceOf[TileAutoAssembler])
        return null;
      return new ContainerAutoAssembler(tile.asInstanceOf[TileAutoAssembler], player);
    }
    //Open painter inventory
    case 4 =>{
      if(!tile.isInstanceOf[TilePainter])
        return null;
      return new ContainerPainter(tile.asInstanceOf[TilePainter], player);
    }
    case i=> return null;
    }
    return null;
  }

  override def getClientGuiElement(ID:Int, player:EntityPlayer, world:World,x:Int, y:Int, z:Int):Object={
    val tile:TileEntity = world.getTileEntity(x, y, z);
    ID match{
    //Open chest inventory
    case 0 =>{
      if(!tile.isInstanceOf[TInventorySimple])
        return null;
      val inv:TInventorySimple = tile.asInstanceOf[TInventorySimple];
      if(inv.isInstanceOf[TSecurable] && inv.asInstanceOf[TSecurable].isSecured())
        return new GuiContainerDynamic(tile, inv.getInventory(), player);
      return new GuiContainerDynamic(inv.getInventory(), player);
    }
    //Open workbench
    case 1 =>{
      return new GuiContainerWorkbench(player, world, x,y,z);
    }
    //Open Worktable inventory
    case 2 =>{
      if(!tile.isInstanceOf[TileWorktable])
        return null;
      return new GuiContainerWorktable(tile.asInstanceOf[TileWorktable], player);
    }
    //Open Auto Assembler inventory
    case 3 =>{
      if(!tile.isInstanceOf[TileAutoAssembler])
        return null;
      return new GuiContainerAutoAssembler(tile.asInstanceOf[TileAutoAssembler], player);
    }
    //Open painter inventory
    case 4 =>{
      if(!tile.isInstanceOf[TilePainter])
        return null;
      return new GuiContainerPainter(tile.asInstanceOf[TilePainter], player);
    }
    case i=> return null;
    }
    return null;
  }

}